src/serialize.h

```diff
@@ -318,22 +318,91 @@ uint64_t ReadCompactSize(Stream& is)
     return nSizeRet;
 }
 
+class CompactSizeFlags {
+private:
+    uint64_t flags;
+public:
+    CompactSizeFlags(uint64_t flagsIn) {
+        if (flagsIn >= 0x100000000ULL && flagsIn <= (uint64_t)MAX_SIZE) {
```

Comment:
Ah, interesting, I never even looked at the value of MAX_SIZE; this logic mirrors the behavior related to picking the bit width in serialization of compact sizes, so it may be that this is *supposed* to be unreachable. I'll double-check.